src/seq/nucleotide.jl

```diff
@@ -966,17 +1118,27 @@ immutable EachKmerIteratorState{T, K}
     nit_state::Int
 end
 
+# Maybe this function should replace the default constructor.
+# Is the (unsafe) default constructor used throughout our code?
```

Comment:
I'm referrint to `eachkmer`. This is could replace the default constructor of `EachKmerIterator` , since it checks `k` and `step`.